app/src/main/java/com/example/androidacademyhomework/UI/movielist/MoviesListFragmentViewModel.kt

```diff
@@ -0,0 +1,15 @@
+package com.example.androidacademyhomework.UI.movielist
+
+import android.app.Application
+import androidx.lifecycle.*
+import com.example.androidacademyhomework.data.Movie
+import com.example.androidacademyhomework.repository.Repository
+
+class MoviesListFragmentViewModel(application: Application) : AndroidViewModel(application) {
+
+    private val repository = Repository(application)
+    val listMoviesLiveData: LiveData<List<Movie>> = liveData {
+        emit(repository.loadMovies())
```

Comment:
Interesting, what scope it will use? What will happen if it gets exception?

Reply:
It will be cancelled until the next launch